### ISSUE TITLE
Bug: Core Grid List is missing some accessibility features

### ISSUE THREAD

**Issue by kavya-b** [SAP/fundamental-ngx#5655]
#### Is this a bug, enhancement, or feature request?
bug/enhancement

#### Briefly describe your proposal.

* [ ] (Core) Grid List, no keyboard arrows support

![image](https://user-images.githubusercontent.com/20265336/150974014-36841f16-6e7e-49c1-be90-aef93990fb74.png)

To support complete a11y in Grid List, we need to use `role="listbox"` and `role="option"` on the list and its items, and follow the a11y guidelines for this role from here:  https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role
However, the current structure does not place the list items under a separate element, and they are siblings with the` title-bar` component; placing `aria-activedescendant` and `aria-multiselectable` on the parent listbox has no effect on the list items. The structure needs to be modified to put the items under another container(which can then serve as a `ul`/listbox).
<img src="https://user-images.githubusercontent.com/53509521/121457995-37189e80-c9c6-11eb-8c28-b001fd82ca80.png" width="400">
#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
latest
#### If this is a bug, please provide steps for reproducing it.
1. Use a screen reader to navigate through Grid List documentation, it does not announce things like multiselectable for the multi-select example, nor are there enough keyboard interactions as specified here:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role#keyboard_interactions

#### Is there anything else we should know?
PR https://github.com/SAP/fundamental-ngx/pull/5643 is handling only the a11y issues pointed out in the issue #4666; since this issue might need structural changes(maybe also introduce breaking changes if a new containing element is introduced for the list items) and can be complicated, a separate issue was raised.
